src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -46,40 +46,30 @@ const DiffExprCompute = (props) => {
     if (hierarchy && hierarchy.length === 0) return;
 
     // If any selected option is deleted, set the option to null
-    Object.keys(comparisonGroup).forEach((type) => {
-      const deleteKeys = {};
-
-      Object.entries(comparisonGroup[type]).forEach(([comparisonKey, selectedCell]) => {
-        selectedCell = getCellSetKey(selectedCell)
-        if (selectedCell && !properties.hasOwnProperty(selectedCell)) deleteKeys[comparisonKey] = null
-      });
-
-      if (Object.keys(deleteKeys).length) {
-
-        dispatch(
-          setComparisonGroup({
-            type,
-            ...comparisonGroup[type],
-            ...deleteKeys,
-          }),
-        );
-      }
+    Object.keys(comparisonGroup).forEach(
```

Comment:
I don't understand too much, but I am guessing that this is dealing with what happens when one of the selected cellSets is deleted, right?

If that is the case, I don't think this should be dealt with here. It should be dealt with the tools we have in redux directly instead of having this go through a different component.

This is effectively dealing with the consequences of a cellSet being deleted, the tool to use for that is a reducer on the `CELL_SETS_DELETE` action. Right now this `useEffect` is performing the work of a reducer in an indirect way.

We trigger the action `CELL_SETS_DELETE`, as a consequence the reducer of each slice should perform its necessary changes.
In this case, I would add a `cellSetsDelete` reducer in the `differentialExpression` slice of the store that checks whether the selected option shoudlb e unselected, and all of this `useEffect` just goes away.

Reply:
Great idea! I'll implement that.

Reply:
implemented